bool CheckPointReached
bool ReverseCheck
bool MovingBack
